lib/datadog/tracing/contrib/aws/service/states.rb

```diff
@@ -16,21 +16,35 @@ def add_tags(span, params)
               execution_arn = params[:execution_arn]
               state_machine_account_id = ''
 
+              if state_machine_arn
+                span.set_tag(Aws::Ext::TAG_STATE_MACHINE_ARN, state_machine_arn)
+                # https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html#API_StartExecution_RequestSyntax:~:text=Required%3A%20No-,stateMachineArn,-The%20Amazon%20Resource
+                # arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName>
+                # arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName>:10
+                # arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName:PROD>
+                # There are 3 patterns to cover and attempt to capture the `myStateMachineName`, it should always be in index 6 and account_id at index 4
+                parts = state_machine_arn.split(':')
+                state_machine_name ||= parts[6]
+                state_machine_account_id = parts[4]
```

Comment:
A bit outside of the scope of this PR, but do you know if `<account-id>` is a sensitive field (that we should be wary of collecting?), like PII?

Reply:
its the aws_account id, we already show that on spans https://app.datadoghq.com/apm/traces?query=%40_top_level%3A1%20%40aws_account%3A850406765696&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&graphType=flamegraph&historicalData=false&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanType=service-entry&start=1695847274250&end=1695848174250&paused=false

so I assume it is not PII but im not sure how to confirm, I can ask/look around our docs

Reply:
we do it for serverless funcs in customer orgs too